Fix `ValidationAction` dispatching of `BotUttered` messages (#11394)
RasaHQ/rasa: changelog/11394.bugfix.md

> Enables the execution of side effects of running custom validation actions such as dispatching bot messages.

Reviewer: ```suggestion
Enables the dispatching of bot messages returned as events by custom actions.
```
